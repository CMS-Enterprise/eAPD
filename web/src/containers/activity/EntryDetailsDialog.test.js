import { shallow } from 'enzyme';
import React from 'react';

import ActivityDialog from './EntryDetailsDialog';

describe('the Activity Dialog component', () => {
  const props = {
    title: 'activity name',
    activityIndex: 2,
    closeModal: jest.fn()
  };

  test('renders correctly', () => {
    const component = shallow(<ActivityDialog {...props} />);
    expect(component).toMatchSnapshot();
  });

  test('it calls closeModal when exiting', () => {
    const component = shallow(<ActivityDialog {...props} />);
    const button = component.dive().find('Button');
    button.simulate('click');
    expect(props.closeModal).toHaveBeenCalled();
  });
});
