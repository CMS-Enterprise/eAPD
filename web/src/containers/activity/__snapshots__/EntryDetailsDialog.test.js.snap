// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Activity Dialog component renders correctly 1`] = `
<Dialog
  ariaCloseLabel="Close modal for activity name"
  className="ds-c-dialog--full"
  closeButtonVariation="transparent"
  closeText="Save and close"
  escapeExitDisabled={false}
  onExit={[Function]}
  title="activity name"
  underlayClickExits={false}
>
  <Tabs>
    <TabPanel
      id="#activity-overview-2"
      selected={false}
      tab="Activity overview"
    >
      <Connect(ActivityOverview)
        activityIndex={2}
      />
      <Connect(StandardsAndConditions)
        activityIndex={2}
      />
    </TabPanel>
    <TabPanel
      id="#activity-goals-2"
      selected={false}
      tab="Goals"
    >
      <Connect(Goals)
        activityIndex={2}
      />
      <Connect(Schedule)
        activityIndex={2}
      />
    </TabPanel>
    <TabPanel
      id="#activity-cost-categories-2"
      selected={false}
      tab="In-house cost categories"
    >
      <Costs
        activityIndex={2}
      />
    </TabPanel>
    <TabPanel
      id="#activity-contractor-costs-2"
      selected={false}
      tab="Private contractor costs"
    >
      <Connect(ContractorResources)
        activityIndex={2}
      />
    </TabPanel>
    <TabPanel
      id="#activity-cost-allocation-2"
      selected={false}
      tab="Cost allocation"
    >
      <Connect(CostAllocate)
        activityIndex={2}
      />
    </TabPanel>
  </Tabs>
</Dialog>
`;
